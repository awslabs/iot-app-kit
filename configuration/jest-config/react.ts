export default {
  setupFilesAfterEnv: ['jest-extended/all'],
  maxWorkers: '50%',
  moduleFileExtensions: ['js', 'ts', 'jsx', 'tsx'],
  collectCoverage: true,
  testEnvironment: 'jsdom',
  transform: {
    '.*\\.(tsx?|jsx?)$': [
      '@swc/jest',
      {
        jsc: {
          transform: {
            react: {
              runtime: 'automatic',
            },
          },
        },
      },
    ],
  },
  transformIgnorePatterns: [],
  moduleNameMapper: {
    '\\.(css|scss)$': 'identity-obj-proxy',
  },
};
