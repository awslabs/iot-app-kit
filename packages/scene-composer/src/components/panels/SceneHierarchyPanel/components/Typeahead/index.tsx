import React, { useCallback } from 'react';
import { Autosuggest } from '@cloudscape-design/components';
import { defineMessages, useIntl } from 'react-intl';

import { useSceneHierarchyData } from '../../SceneHierarchyDataProvider';

const strings = defineMessages({
  enteredTextLabel: {
    defaultMessage: 'Use: {value}',
    description: 'Label clarifying the exact text a user is searching for.',
  },
});

const Typeahead = (props) => {
  const { searchTerms, search } = useSceneHierarchyData();
  const { formatMessage } = useIntl();

  const doSearch = useCallback(
    (e) => {
      const filter = e.detail.value;
      search(filter);
    },
    [search],
  );

  return (
    <Autosuggest
      value={searchTerms}
      onChange={doSearch}
      enteredTextLabel={(val) => formatMessage(strings.enteredTextLabel, { value: val })}
      {...props}
    />
  );
};

export default Typeahead;
