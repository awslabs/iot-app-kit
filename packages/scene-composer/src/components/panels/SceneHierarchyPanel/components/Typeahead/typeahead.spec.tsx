import React from 'react';
import { render } from '@testing-library/react';

import Typeahead from '.';

jest.mock('@awsui/components-react/autosuggest', () => (props) => <div data-mocked={'Autosuggest'} {...props} />);

describe('<Typeahead />', () => {
  it('should match snapshot', () => {
    const { container } = render(<Typeahead />);
    expect(container).toMatchSnapshot();
  });
});
