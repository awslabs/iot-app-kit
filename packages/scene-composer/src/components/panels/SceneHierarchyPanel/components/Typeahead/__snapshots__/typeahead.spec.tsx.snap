// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typeahead /> should match snapshot 1`] = `
<div>
  <div
    data-mocked="Autosuggest"
    value=""
  />
</div>
`;
